bigquery_etl/view/__init__.py

```diff
@@ -291,6 +280,30 @@ def has_changes(self, target_project=None):
                 print(f"view {target_view_id} will change: labels do not match")
                 return True
 
+        schema_file = Path(self.path).parent / "schema.yaml"
+        table_schema = Schema.from_bigquery_schema(table.schema)
+        # check schema based on schema file
+        if schema_file.is_file():
+            view_schema = Schema.from_schema_file(schema_file)
+        else:  # check schema based on dry run results
+            try:
+                query_job = client.query(
+                    query=self.content,
+                    job_config=bigquery.QueryJobConfig(
+                        dry_run=True, use_legacy_sql=False
+                    ),
+                )
+                view_schema = Schema.from_bigquery_schema(query_job.schema)
+            except Forbidden:
+                print(
+                    f"Missing permission to dry run view {target_view_id} to get schema"
+                )
+                view_schema = None
+
+        if view_schema is not None and not view_schema.equal(table_schema):
```

Comment:
How do descriptions factor in here? If the view is deployed with descriptions, will comparing schemas w/ the dryrun schema return False?

Reply:
Do you mean view-level or column descriptions?  For view descriptions, they're compared in the `if self.metadata is not None:` section above so schemas are only compared if view descriptions match.  
For column descriptions, my understanding is the `.equal` method here ignores them so this will only return true if there are different fields.  I actually didn't consider that when writing this since my intention was to keep the behaviour as similar to before as possible.

Reply:
I meant schema descriptions here. Might be worth a test to double-check, otherwise we might get schemas that always "have changes" because the dryrun schema won't contain those descriptions

Reply:
I can confirm descriptions in the deployed schema don't cause the view to be published.  
For example, running `print(f"{view.name}: {view.has_changes()}")` with the `airflow_task_fail` view outputs:
```
Warning: descriptions for root.dag_id differ
Warning: descriptions for root.task_id differ
airflow_task_fail: False
```